src/Block.php

```diff
@@ -34,26 +63,46 @@ public function channelLinks() {
     foreach ($enabled_channels as $channel_name) {
       $channel_options = isset($options[$channel_name]) ? $options[$channel_name] : array();
       $channel = $loader->channel($channel_name, $this, $channel_options);
-      if($channel->enabled() && ($channel_link = $channel->render())) {
+      if ($channel->enabled() && ($channel_link = $channel->render())) {
         $links[$channel_name] = $channel_link;
       }
     }
     return $links;
   }
 
+  /**
+   * Returns the absolute URL pointing to the node that shall be shared.
+   */
   public function getUrl($absolute = TRUE) {
     $link = $this->options['link'];
     return url($link['path'], $link + array('absolute' => $absolute));
   }
 
+  /**
+   * Returns the relative URL pointing to the node that shall be shared.
```

Comment:
It’s not always a node that is being shared. It can be an arbitrary internal path or external URL.